### PR TITLE
fix (Core/Scripting)  SMART_EVENT_NEAR_PLAYERS 

### PR DESCRIPTION
**Pull request by skelUA**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -4367,10 +4367,10 @@ void SmartScript::ProcessEvent(SmartScriptHolder& e, Unit* unit, uint32 var0, ui
                 {
                     if (IsPlayer(target))
                         playerCount++;
-
-                    if (playerCount >= e.event.nearPlayer.minCount)
-                        ProcessAction(e, target->ToUnit());
                 }
+                    if (playerCount >= e.event.nearPlayer.minCount)
+                        ProcessAction(e, unit);
+
```
Comment (Kitzunu) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:4373]: Several things here.

1. What was broken in the first place?
2. By doing this you are making the for loop above do nothing
3. The ident is wrong

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by skelUA** [comment]
@Kitzunu  plz review

**Comment by Kitzunu** [comment]
My review still stands. You have not adressed any of the concerns I lifted

**Comment by skelUA** [comment]
> My review still stands. You have not addressed any of the concerns I lifted. You can't just simply resolve a review without doing anything and expect me to forget it lol.

But i answered on all your questions. Or what i do wrong?

image

**Comment by Kitzunu** [comment]
the comment is set as "pending" you need to press these buttons 
image

**Review by skelUA — COMMENTED** [review]
add comment

**Comment by pangolp** [comment]
What status is this pull request in?

**Comment by skelUA** [comment]
@Kitzunu So may you review now or i need to do something else ?

**Pull request closed without merging**